tests/shared/src/test/scala/io/circe/generic/SemiautoDerivedSuite.scala

```diff
@@ -48,7 +48,7 @@ class SemiautoDerivedSuite extends CirceSuite {
     implicit val encodeRecursiveAdtExample: Encoder[RecursiveAdtExample] = deriveEncoder
   }
 
-  case class RecursiveWithOptionExample(o: Option[RecursiveWithOptionExample])
+  @JsonCodec case class RecursiveWithOptionExample(o: Option[RecursiveWithOptionExample])
```

Comment:
I think I'd prefer to have a new case class (or possibly even a new suite?) rather than change existing tests.